### PR TITLE
feat(api): limit patient create sandbox

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport-internal#764

Ref: 764

### Description

Limit patient create in sandbox mode

### Release Plan

- [ ] Merge once approved

### CODE REVIEW

File: api/app/src/routes/medical/patient.ts [hunk 1]
```diff
@@ -50,6 +52,16 @@ router.post(
     const facilityId = getFromQueryOrFail("facilityId", req);
     const payload = patientCreateSchema.parse(req.body);
 
+    if (Config.isSandbox()) {
+      // limit the amount of patients that can be created in sandbox mode
+      const numPatients = await Patient.count({ where: { cxId } });
+      if (numPatients >= Config.SANDBOX_PATIENT_LIMIT) {
+        return res.status(status.BAD_REQUEST).json({
+          message: `Cannot create more than ${Config.SANDBOX_PATIENT_LIMIT} patients in Sandbox mode!`,
+        });
```
Comment (leite08) [inline api/app/src/routes/medical/patient.ts:61]: We should probably process this on the Dash and display a user friendly message there, right? Assuming the UI will be available on `sandbox`, of course.

File: api/app/src/routes/medical/patient.ts [hunk 2]
```diff
@@ -50,6 +52,16 @@ router.post(
     const facilityId = getFromQueryOrFail("facilityId", req);
     const payload = patientCreateSchema.parse(req.body);
 
+    if (Config.isSandbox()) {
+      // limit the amount of patients that can be created in sandbox mode
+      const numPatients = await Patient.count({ where: { cxId } });
+      if (numPatients >= Config.SANDBOX_PATIENT_LIMIT) {
```
Comment (Goncharo) [inline api/app/src/routes/medical/patient.ts:58]: should this be `numPatients > Config.SANDBOX_PATIENT_LIMIT`?

otherwise it will error on 20

**Review by Goncharo — CHANGES_REQUESTED** [review]
minor logic issue, also consider @leite08's comment

**Review by Goncharo — APPROVED** [review]
:shipit:

**Pull request merged**